patina_mm: Include readme in module docs (#1139)
OpenDevicePartnership/patina: components/patina_mm/README.md

  fn init(&self) -> patina::error::Result<()> {
    /* platform MMI init code */
    Ok(())
  }
}

struct ExamplePlatform;

impl ComponentInfo for ExamplePlatform {
  fn configs(mut add: Add<Config>) {
    // See `MmCommunicationConfiguration` struct for configuration options
    add.config(patina_mm::config::MmCommunicationConfiguration {
      acpi_base: patina_mm::config::AcpiBase::Mmio(0x0), // Actual ACPI base address will be set during boot
      cmd_port: patina_mm::config::MmiPort::Smi(0xB2),
      data_port: patina_mm::config::MmiPort::Smi(0xB3),
      enable_comm_buffer_updates: false,
      updatable_buffer_id: None,
      comm_buffers: vec![],
    });
  }

  fn components(mut add: Add<Component>) {
    add.component(patina_mm::component::sw_mmi_manager::SwMmiManager::new());
    add.component(patina_mm::component::communicator::MmCommunicator::new());
  }

  fn services(mut add: Add<Service>) {
    // An optional service to enable platform MM. Since it has no dependencies, we register the service directly. If it
    // had dependencies, This would be a component instead.
    add.service(ExamplePlatformMmControl::default());
  }
}
```

## Service Usage guidance

> Below is example usage of the `MmCommunication` service for component writers who wish to consume and use the

Reviewer: ```suggestion
Below is example usage of the `MmCommunication` service for component writers who wish to use this 

```

  fn init(&self) -> patina::error::Result<()> {
    /* platform MMI init code */
    Ok(())
  }
}

struct ExamplePlatform;

impl ComponentInfo for ExamplePlatform {
  fn configs(mut add: Add<Config>) {
    // See `MmCommunicationConfiguration` struct for configuration options
    add.config(patina_mm::config::MmCommunicationConfiguration {
      acpi_base: patina_mm::config::AcpiBase::Mmio(0x0), // Actual ACPI base address will be set during boot
      cmd_port: patina_mm::config::MmiPort::Smi(0xB2),
      data_port: patina_mm::config::MmiPort::Smi(0xB3),
      enable_comm_buffer_updates: false,
      updatable_buffer_id: None,
      comm_buffers: vec![],
    });
  }

  fn components(mut add: Add<Component>) {
    add.component(patina_mm::component::sw_mmi_manager::SwMmiManager::new());
    add.component(patina_mm::component::communicator::MmCommunicator::new());
  }

  fn services(mut add: Add<Service>) {
    // An optional service to enable platform MM. Since it has no dependencies, we register the service directly. If it
    // had dependencies, This would be a component instead.
    add.service(ExamplePlatformMmControl::default());
  }
}
```

## Service Usage guidance

> Below is the integration guidance for component writers who wish to consume and use the `MmCommunication` and

Reviewer 2: I'm not opposed to this going into the crate documentation, but my original plan for this level of detail was to defer to the code in patina-dxe-core-qemu. If a simple link there would accomplish what you're looking for, it would reduce our overall maintenance burden, and I think roughly accomplish the same thing.

> Below is the integration guidance for component writers who wish to consume and use the `MmCommunication` and

Author: I think an example would be beneficial here. But I agree it does not necessarily need to be this complex.

With that said, I pulled this example from the crate module documentation, not from patina-dxe-core:

https://github.com/OpenDevicePartnership/patina/blob/0914dd88daaac25105ae94b44fb34228b7880a42/components/patina_mm/src/lib.rs#L34

I'd be happy to reduce the complexity of the example if you prefer - or, as you requested, provide a link to the test code

> Below is the integration guidance for component writers who wish to consume and use the `MmCommunication` and

Reviewer 2: I'm personally fine with a link. I'd almost want to direct people more toward the Q35 code because it's still works as an example and it is more tangible in that it will actually perform working communication calls and someone looking at MM code is likely interested in that.

> Below is the integration guidance for component writers who wish to consume and use the `MmCommunication` and

Author: I will create a really simple example for the module documentation and provide a link to patina-dxe-core-qemu
